Can't build servant-swagger

`servant-swagger`'s custom setup needs the `Cabal` library but doesn't list it in `setup-depends`:

https://github.com/haskell-servant/servant/blob/c388c5e82c3df5d8ae63b523fea012c23e37410a/servant-swagger/servant-swagger.cabal#L50-L53

This causes a build error:

```
/stack/.stack/setup-exe-src/setup-shim-mPHDZzAJ.hs:3:1: error:        
    Could not load module ‘Distribution.PackageDescription’        
    It is a member of the hidden package ‘Cabal-3.2.1.0’.        
    You can run ‘:set -package Cabal’ to expose it.  
    (Note: this unloads all the modules in the current scope.)        
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.        
  |        
3 | import Distribution.PackageDescription (PackageDescription, emptyHookedBuildInfo)        
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  
```

The package description from the old repo correctly lists a dependency on `Cabal`:

https://github.com/haskell-servant/servant-swagger/blob/7ac475a40b4b95a50bc2094f9461c62bd7fbb1fd/servant-swagger.cabal#L50-L54

--- a/servant-swagger/servant-swagger.cabal
+++ b/servant-swagger/servant-swagger.cabal
@@ -92,7 +92,7 @@ test-suite doctests
   build-depends:
     base,
     directory >= 1.0,
-    doctest >= 0.18 && <0.19,
+    doctest >= 0.17 && <0.19,
     servant,
     QuickCheck,
     filepath

--- a/servant-swagger/servant-swagger.cabal
+++ b/servant-swagger/servant-swagger.cabal
@@ -50,6 +50,7 @@ source-repository head
 custom-setup
   setup-depends:
     base >=4.9 && <5,
+    Cabal >= 1.24 && < 3.5,
     cabal-doctest >=1.0.6 && <1.1
 
 library

--- a/stack.yaml
+++ b/stack.yaml
@@ -11,6 +11,7 @@ packages:
 - servant-conduit
 - servant-machines/
 - servant-pipes/
+- servant-swagger/
 
 # allow-newer: true # ignores all bounds, that's a sledgehammer
 # - doc/tutorial/

